feat/add-work-calendar-deployment draft (#131)
libremfg/rhize-docs: content/deploy/install/services.md

$ helm upgrade --install router -f router.yaml {{< param application_name >}}/router -n {{< param application_name >}}
```

## Optional: Calendar Service (Draft)

> The {{< param brand_name >}} calendar service monitors work calendar definitions and creates work calendar entries in real time, both in the [Graph DB](#db) and Timeseries DB.

Reviewer: ```suggestion
The {{< param brand_name >}} calendar service monitors work calendar definitions and creates work calendar entries in real time, both in the [Graph](#db) and time-series databases.
```

> **Requirements:** The calendar service requires the [GraphDB](#db), [Keycloak](#keycloak), and [NATS](#nats) services.

> > **Note:** It is a prerequisite to have a Timeseries DB installed such as [InfluxDB](https://influxdata.com/), [QuestDB](https://questdb.io) or [TimescaleDB](https://www.timescale.com/). The following instructions are specific to QuestDB.

Reviewer: ```suggestion
{{% notice "note" %}}
It is a prerequisite to have a time-series DB installed such as [InfluxDB](https://influxdata.com/), [QuestDB](https://questdb.io) or [TimescaleDB](https://www.timescale.com/). The following instructions are specific to QuestDB.
{{% /notice %}}
```

Reviewer: Shortcodes and other stuff like this are documented in [CONTRIBUTING.MD](https://github.com/libremfg/rhize-docs/blob/main/CONTRIBUTING.md#notices)

Install the calendar service with these steps:

> 1. Create tables in timeseries

Reviewer: ```suggestion
1. Create tables in the time series: 
```

```sql
CREATE TABLE IF NOT EXISTS PSDT_POT(
  EquipmentId SYMBOL,
  EquipmentVersion STRING,

>   WorkCalendarId STRING,
>   WorkCalendarIid STRING,
>   WorkCalendarDefinitionId STRING,
>   WorkCalendarDefinitionEntryId STRING,
>   WorkCalendarDefinitionEntryIid STRING,
>   WorkCalendarEntryId STRING,
>   WorkCalendarEntryIid SYMBOL,
>   HierarchyScopeId STRING,
>   EntryType STRING,
>   ISO22400CalendarState STRING,
>   isDeleted boolean,
>   updatedAt TIMESTAMP,
>   time TIMESTAMP,
>   lockerCount INT,
>   lockers STRING
> ) TIMESTAMP(time) PARTITION BY month
> DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);
> 
> CREATE TABLE IF NOT EXISTS PDOT_PBT(
>   EquipmentId SYMBOL,
>   EquipmentVersion STRING,
>   WorkCalendarId STRING,
>   WorkCalendarIid STRING,
>   WorkCalendarDefinitionId STRING,
>   WorkCalendarDefinitionEntryId STRING,
>   WorkCalendarDefinitionEntryIid STRING,
>   WorkCalendarEntryId STRING,
>   WorkCalendarEntryIid SYMBOL,
>   HierarchyScopeId STRING,
>   EntryType STRING,
>   ISO22400CalendarState STRING,
>   isDeleted boolean,
>   updatedAt TIMESTAMP,
>   time TIMESTAMP,
>   lockerCount INT,
>   lockers STRING
> ) TIMESTAMP(time) PARTITION BY month
> DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);
> 
> CREATE TABLE IF NOT EXISTS Calendar_AdHoc(
>   EquipmentId SYMBOL,
>   EquipmentVersion STRING,
>   WorkCalendarId STRING,
>   WorkCalendarIid STRING,
>   WorkCalendarDefinitionId STRING,
>   WorkCalendarDefinitionEntryId STRING,
>   WorkCalendarDefinitionEntryIid STRING,
>   WorkCalendarEntryId STRING,
>   WorkCalendarEntryIid SYMBOL,
>   HierarchyScopeId STRING,
>   EntryType STRING,
>   ISO22400CalendarState STRING,
>   isDeleted boolean,
>   updatedAt TIMESTAMP,
>   time TIMESTAMP,
>   lockerCount INT,
>   lockers STRING
> ) TIMESTAMP(time) PARTITION BY month
> DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);
> ```

Reviewer: ```suggestion
    ```sql
    CREATE TABLE IF NOT EXISTS PSDT_POT(
      EquipmentId SYMBOL,
      EquipmentVersion STRING,
      WorkCalendarId STRING,
      WorkCalendarIid STRING,
      WorkCalendarDefinitionId STRING,
      WorkCalendarDefinitionEntryId STRING,
      WorkCalendarDefinitionEntryIid STRING,
      WorkCalendarEntryId STRING,
      WorkCalendarEntryIid SYMBOL,
      HierarchyScopeId STRING,
      EntryType STRING,
      ISO22400CalendarState STRING,
      isDeleted boolean,
      updatedAt TIMESTAMP,
      time TIMESTAMP,
      lockerCount INT,
      lockers STRING
    ) TIMESTAMP(time) PARTITION BY month
    DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);

    CREATE TABLE IF NOT EXISTS PDOT_PBT(
      EquipmentId SYMBOL,
      EquipmentVersion STRING,
      WorkCalendarId STRING,
      WorkCalendarIid STRING,
      WorkCalendarDefinitionId STRING,
      WorkCalendarDefinitionEntryId STRING,
      WorkCalendarDefinitionEntryIid STRING,
      WorkCalendarEntryId STRING,
      WorkCalendarEntryIid SYMBOL,
      HierarchyScopeId STRING,
      EntryType STRING,
      ISO22400CalendarState STRING,
      isDeleted boolean,
      updatedAt TIMESTAMP,
      time TIMESTAMP,
      lockerCount INT,
      lockers STRING
    ) TIMESTAMP(time) PARTITION BY month
    DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);

    CREATE TABLE IF NOT EXISTS Calendar_AdHoc(
      EquipmentId SYMBOL,
      EquipmentVersion STRING,
      WorkCalendarId STRING,
      WorkCalendarIid STRING,
      WorkCalendarDefinitionId STRING,
      WorkCalendarDefinitionEntryId STRING,
      WorkCalendarDefinitionEntryIid STRING,
      WorkCalendarEntryId STRING,
      WorkCalendarEntryIid SYMBOL,
      HierarchyScopeId STRING,
      EntryType STRING,
      ISO22400CalendarState STRING,
      isDeleted boolean,
      updatedAt TIMESTAMP,
      time TIMESTAMP,
      lockerCount INT,
      lockers STRING
    ) TIMESTAMP(time) PARTITION BY month
    DEDUP UPSERT KEYS(time, EquipmentId, WorkCalendarEntryIid);
    ```
```

Reviewer: This indention is to prevent the numbering from resetting

> 2. Modify the calendar YAML file as needed.
> 
> 3. Deploy with helm 

Reviewer: ```suggestion
1. Modify the calendar YAML file as needed.

1. Deploy with helm 
```

Reviewer: The parser uses the AST to set numbers (see last comment), so keeping them all at 1 makes it easier to move steps around.

> 3. Deploy with helm 

Reviewer: ```suggestion
3. Deploy with Helm:
```

> ```bash
> helm install calendar-service -f calendar-service.yaml {{< param application_name >}}/calendar-service -n {{< param application_name >}}
> ```

Reviewer: ```suggestion
    ```bash
    helm install calendar-service -f calendar-service.yaml {{< param application_name >}}/calendar-service -n {{< param application_name >}}
    ```
```
